lxd/instance.go

```diff
@@ -927,3 +930,36 @@ func containerDetermineNextSnapshotName(d *Daemon, c instance.Instance, defaultP
 
 	return pattern, nil
 }
+
+var instanceDriversCacheVal atomic.Value
+var instanceDriversCacheLock sync.Mutex
+
+func readInstanceDriversCache() map[string]string {
+	drivers := instanceDriversCacheVal.Load()
+	if drivers == nil {
+		createInstanceDriversCache()
+		drivers = instanceDriversCacheVal.Load()
+	}
+
+	return drivers.(map[string]string)
+}
+
+func createInstanceDriversCache() {
+	// Create the list of instance drivers in use on this LXD instance
+	// namely LXC and QEMU. Given that LXC and QEMU cannot update while
+	// the LXD instance is running, only one cache is ever needed.
+
+	data := map[string]string{}
+
+	info := drivers.SupportedInstanceDrivers()
+	for _, entry := range info {
+		data[entry.Name] = entry.Version
```

Comment:
Should make this those with empty. `entry.Version`

Reply:
Fixed in 1dbbe99